### PR TITLE
Script paint worklets speculative evaluation

### LINKED ISSUE #17486: Paint worklets: speculatively draw a paint image
During styling, we know the computed values for properties, so we can speculatively draw the paint image, assuming that the size hasn't changed.

### PR DESCRIPTION
**Pull request by asajeffrey**
This PR speculatively calls paint worklets during style, which increases the concurrency, since it increases the chance that the cache will have the right result in it when it comes to layout. The speculation is wasted effort if the size of the element has changed, but this is often not the case.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #17486 and #17369.
- [X] These changes do not require tests because it's a performance improvement

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17810)

### CODE REVIEW

**Comment by asajeffrey** [comment]
This PR is dependent on #17634.

**Comment by asajeffrey** [comment]
#17634 landed, so this PR is ready for review!

File: components/style/traversal.rs [hunk 1]
```diff
@@ -520,6 +522,8 @@ where
         }
     }
 
+    notify_paint_worklet(context, data);
```
Comment (emilio) [inline components/style/traversal.rs]: Are you sure you want this outside of the `if compute_self` block? Do we really always need to paint it even if the style is the same?

**Review by emilio — APPROVED** [review]
Looks fine, just a question.

**Comment by emilio** [comment]
Also, do you have any numbers on this?

File: components/style/traversal.rs (hunk 1 above)
Comment (asajeffrey) [inline components/style/traversal.rs]: Good point. Fixed.

**Comment by asajeffrey** [comment]
@emilio I'm afraid not, to get decent numbers we'd need a page with reasonably complex style/layout and a paint worklet. All the paint worklet demos are quite simple at the moment :/

@bors-servo r=emilio

**Comment by asajeffrey** [comment]
No actual numbers, but at least the cache is doing its job!
```
$ RUST_LOG=script::dom::paintworkletglobalscope ./mach run -d --pref=dom.worklet.enabled http://googlechrome.github.io/houdini-samples/
DEBUG:script::dom::paintworkletglobalscope: Creating paint worklet global scope for pipeline (0,2).
DEBUG:script::dom::paintworkletglobalscope: Registering paint image name ripple.
DEBUG:script::dom::paintworkletglobalscope: Registering definition ripple.
DEBUG:script::dom::paintworkletglobalscope: Creating paint worklet global scope for pipeline (0,2).
DEBUG:script::dom::paintworkletglobalscope: Registering paint image name ripple.
DEBUG:script::dom::paintworkletglobalscope: Registering definition ripple.
DEBUG:script::dom::paintworkletglobalscope: Creating paint worklet global scope for pipeline (0,2).
DEBUG:script::dom::paintworkletglobalscope: Registering paint image name ripple.
DEBUG:script::dom::paintworkletglobalscope: Registering definition ripple.
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(0,0) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Cache miss on paint worklet ripple!
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(300,300) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(300,300) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Cache hit on paint worklet ripple!
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(300,300) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Cache hit on paint worklet ripple!
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(300,300) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Cache hit on paint worklet ripple!
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(300,300) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Cache hit on paint worklet ripple!
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(300,300) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Cache hit on paint worklet ripple!
DEBUG:script::dom::paintworkletglobalscope: Cache hit on paint worklet ripple!
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(300,300) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Cache hit on paint worklet ripple!
DEBUG:script::dom::paintworkletglobalscope: Invoking a paint callback ripple(300,300) at 1.
DEBUG:script::dom::paintworkletglobalscope: Invoking paint function ripple.
DEBUG:script::dom::paintworkletglobalscope: Cache hit on paint worklet ripple!
```
@bors-servo r=emilio

**Pull request merged**